src/Core/Domain/Order/ValueObject/OrderDetailRefund.php

```diff
@@ -0,0 +1,143 @@
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2020 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Domain\Order\ValueObject;
+
+use PrestaShop\PrestaShop\Core\Domain\Order\Exception\EmptyRefundAmountException;
+use PrestaShop\PrestaShop\Core\Domain\Order\Exception\EmptyRefundQuantityException;
+use PrestaShop\PrestaShop\Core\Domain\Order\Exception\OrderException;
+
+/**
+ * Class ProductRefund
+ */
+class OrderDetailRefund
+{
+    /**
+     * @var int
+     */
+    private $orderDetailId;
+
+    /**
+     * @var int
+     */
+    private $productQuantity;
+
+    /**
+     * @var float|null
+     */
+    private $refundAmount;
+
+    /**
+     * @param int $orderDetailId
+     * @param int $productQuantity
+     * @param float $refundAmount
+     *
+     * @return self
+     *
+     * @throws EmptyRefundAmountException
+     * @throws OrderException
+     */
+    public static function createPartialRefund(int $orderDetailId, int $productQuantity, float $refundAmount): self
+    {
+        if (0 >= $refundAmount) {
+            throw new EmptyRefundAmountException();
+        }
+
+        return new self($orderDetailId, $productQuantity, $refundAmount);
+    }
+
+    /**
+     * @param int $orderDetailId
+     * @param int $productQuantity
+     *
+     * @return self
+     *
+     * @throws OrderException
+     */
+    public static function createStandardRefund(int $orderDetailId, int $productQuantity): self
+    {
+        return new self($orderDetailId, $productQuantity, null);
+    }
+
+    /**
+     * @param int $orderDetailId
+     * @param int $productQuantity
+     * @param float|null $refundAmount
+     *
+     * @throws OrderException
+     */
+    private function __construct(int $orderDetailId, int $productQuantity, ?float $refundAmount)
+    {
+        $this->assertIntegerIsGreaterThanZero($orderDetailId);
+        if (0 >= $productQuantity) {
+            throw new EmptyRefundQuantityException();
+        }
+        $this->orderDetailId = $orderDetailId;
+        $this->productQuantity = $productQuantity;
+        $this->refundAmount = $refundAmount;
+    }
+
+    /**
+     * @return int
+     */
+    public function getOrderDetailId(): int
+    {
+        return $this->orderDetailId;
+    }
+
+    /**
+     * @return int
+     */
+    public function getProductQuantity(): int
+    {
+        return $this->productQuantity;
+    }
+
+    /**
+     * @return float|null
+     */
+    public function getRefundAmount(): ?float
+    {
+        return $this->refundAmount;
+    }
+
+    /**
+     * @param int $orderDetailId
+     *
+     * @throws OrderException
+     */
+    private function assertIntegerIsGreaterThanZero(int $orderDetailId)
```

Comment:
```suggestion
    private function assertOrderDetailIdsGreaterThanZero(int $orderDetailId)
```
if we look the code, you're not checking for integer, but for `$orderDetailId` (same in the exception message)

Reply:
Done